zkvm/src/vm.rs

```diff
@@ -579,6 +583,37 @@ where
         })
     }
 
+    fn delegate(&mut self) -> Result<(), VMError> {
+        // Signature
+        let sig = self.pop_item()?.to_data()?.to_bytes();
+        let signature = Signature::from_bytes(SliceReader::parse(&sig, |r| r.read_u8x64())?)?;
+
+        // Program
+        let prog = self.pop_item()?.to_data()?;
+        let prog_bytes = prog.clone().to_bytes();
+
+        // Place all items in payload onto the stack
+        let contract = self.pop_item()?.to_contract()?;
+        for item in contract.payload.into_iter() {
+            self.push_item(item);
+        }
+
+        // Verification key from predicate
+        let verification_key = contract.predicate.to_key()?;
+
+        // Verify signature using Verification key, over the message `program`
+        let mut t = Transcript::new(b"ZkVM.delegate");
+        t.commit_bytes(b"prog", &prog_bytes);
```

Comment:
let's avoid extra variables (`prog_bytes`) to make the logic cleaner:
```suggestion
        t.commit_bytes(b"prog", &prog.clone().to_bytes());
```